### ISSUE TITLE
[Feature Request] Alphabetically sort `typing.Literal` types

### ISSUE THREAD

**Issue by michaeloliverx** [astral-sh/ruff#7220]
I want the ability to alphabetically `Literal` types e.g.

```python
from typing import Literal

# before
ErrorCode = Literal[
    "foo",
    "bar",
]

# after
ErrorCode = Literal[
    "bar",
    "foo",
]
```

This is related to https://github.com/astral-sh/ruff/issues/2315

I think having the ability to sort a range of things such as enums, literal collections, typed dict keys would be great. The rules should probably live in the `RUF` ruleset and perhaps be enabled / disabled via an action comment.

**Comment by charliermarsh** [comment]
Makes sense, very similar to #1198 -- a different case of the same idea.
